src/array.rs

```diff
@@ -62,6 +199,20 @@ pub type WriteOnly<T, M> = Generic<T, M, marker::WriteOnly>;
 /// a.update_volatile_at(0, |v| {
 ///     *v *= 2;
 /// });
+///
+/// // Below are the equivalent examples using `.at()` method.
+///
+/// // Read the 3rd element of the array.
+/// a.at(3).read_volatile();
+///
+/// // Write 42 as the 5th element of the array.
+/// a.at(5).write_volatile(42);
+///
+/// // Update the 0th element.
+/// a.at(0).update_volatile(|v| {
+///     *v *= 2;
+/// })
+///
```

Comment:
This is an example of indexing using `.at(i)` methods. Unlike `.set_at(i)`, this doesn't require to refer any traits.

Basically, `.at(i)` method is introduced to replace `.***_volatile_at(i)` with `.at(i).***_volatile()` so most use cases should be limited to these situations.